File: src/main/java/com/thebluehats/server/core/modules/RegionManagerModule.java
package com.thebluehats.server.core.modules;

import com.google.inject.AbstractModule;
import com.google.inject.Provides;
import com.google.inject.Singleton;
import com.thebluehats.server.game.managers.world.regionmanager.RegionManager;

public class RegionManagerModule extends AbstractModule {
    @Provides
    @Singleton
    static RegionManager provideRegionManager() {
        return new RegionManager();
    }

    @Override
    protected void configure() {
    }
}

File: minecraft-server/src/main/kotlin/com/thebluehats/server/core/modules/RegionManagerModule.kt
package com.thebluehats.server.core.modules

import com.google.inject.AbstractModule
import com.google.inject.Provides
import com.google.inject.Singleton
import com.thebluehats.server.game.managers.world.regionmanager.RegionManager

class RegionManagerModule : AbstractModule() {
    override fun configure() {}

    companion object {
        @Provides
        @Singleton
        fun provideRegionManager(): RegionManager {
            return RegionManager()
        }
    }
}